app/code/Magento/Ui/DataProvider/SearchResultFactory.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Ui\DataProvider;
+
+use Magento\Framework\Api\AttributeValueFactory;
+use Magento\Framework\Api\Search\DocumentFactory;
+use Magento\Framework\Api\Search\SearchResultInterface;
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\Framework\Api\Search\SearchResultFactory as BaseSearchResultFactory;
+use Magento\Framework\EntityManager\HydratorInterface;
+
+/**
+ * Helps to use Repositories (instead of Collections) in UI Components Data providers
```

Comment:
Helps or allows? (i.e. do we need some additional code to make it workable)

Reply:
Fixed (no additional code is needed)